test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java

```diff
@@ -897,4 +900,61 @@ public void txnCanCommitIfOverwrittenAndRetry() throws Exception {
         assertThat(get("k1")).isEqualTo("v1");
         assertThat(get("k2")).isEqualTo("v2");
     }
+
+    @Test
+    public void snapshotReadBeforeCompactionMark() {
+        final int entryNum = RECORDS_PER_SEGMENT + 1;
+
+        t(1, () -> put("k" , "v0"));    // TS = 0
+        t(1, () -> put("k" , "v1"));    // TS = 1
+        t(1, () -> put("k" , "v2"));    // TS = 2
+
+        t(2, this::TXBegin);
+        t(2, () -> get("k"));           // thread2 syncs to TS = 2
+
+        final int timestamp = 3;
+        for (int i = timestamp; i < entryNum; ++i) {
+            AtomicInteger version = new AtomicInteger(i);
+            t(1, () -> put("k" , "v" + version.get()));
+        }
+
+        // Update committed tail so that compactor can run.
+        Utils.updateCommittedTail(getRuntime().getLayoutView().getLayout(), getRuntime(), entryNum - 1);
+
+        // run compaction
+        t(1, () -> startCompaction(getRuntime(), getLogUnit(SERVERS.PORT_0)));
+        t(2, () -> get("k"))
+                .assertThrows().hasCauseInstanceOf(TrimmedException.class);
+        t(2, () -> assertThat(getRuntime().getAddressSpaceView().getCompactionMark().get())
+                        .isEqualTo(RECORDS_PER_SEGMENT));
+        t(2, this::TXEnd);
+    }
+
+    @Test
+    public void snapshotReadAfterCompactionMark() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651450)